src/database/datatools/constructions.py

```
@@ -0,0 +1,214 @@
+                        layer19: Optional[str] = None,
+                        layer20: Optional[str] = None) -> None:
+    
+    conn = sqlite3.connect(db_path)
+    conn.row_factory = sqlite3.Row
+    cursor = conn.cursor()
+
+    cursor.execute("SELECT * FROM constructions WHERE id=?", (construction_id,))
+    row = cursor.fetchone()
+    if not row:
+        conn.close()
+        raise ValueError(f"No construction found with id {construction_id}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)